src/tactic/sort.lean

```diff
@@ -0,0 +1,158 @@
+  e_eq ← mk_app `eq [e, e'],
+  n ← get_unused_name,
+  assert n e_eq,
+  e_eq_fmt ← pp e_eq,
+  reflexivity <|>
+    `[{ simp only [add_comm, add_assoc, add_left_comm], done, }] <|>
+    -- `[{ abel, done, }] <|> -- this works too. it's more robust but also a bit slower
+      fail format!"Failed to prove:\n {e_eq_fmt}",
+  h ← get_local n,
+  match hyp with
+  | some loc := do
+    get_local loc >>= rewrite_hyp h,
+    tactic.clear h
+  | none     := do
+    rewrite_target h,
+    tactic.clear h
+  end
+| [] := skip
+end
+
+/-- Partially traverses an expression in search for a sum of terms.
+When `recurse_on_expr` finds a sum, it sorts it using `sorted_sum_with_rel`. -/
+meta def recurse_on_expr (t : list expr) (hyp : option name) (rel : expr → expr → bool) :
+  expr → tactic unit
+| e@`(%%_ + %%_)     := sorted_sum_with_rel hyp rel t e
+| (expr.lam _ _ _ e) := recurse_on_expr e
+| (expr.pi  _ _ _ e) := recurse_on_expr e
+| e                  := e.get_app_args.mmap' recurse_on_expr
+
+/-- Calls `recurse_on_expr` with the right expression, depending on the tactic location. -/
+meta def sort_summands_aux (rel : expr → expr → bool) (t : list expr) : option name → tactic unit
+| (some hyp) := get_local hyp >>= infer_type >>= recurse_on_expr t hyp rel
+| none       := target >>= recurse_on_expr t none rel
+
+end with_rel
+
+/--  The order on `polynomial.monomial n r`, where monomials are compared by their "exponent" `n`.
+If the expression is not a monomial, then the weight is `⊥`. -/
+meta def monomial_weight (e : expr) : option ℕ :=
+match e.app_fn with
+| `(coe_fn $ polynomial.monomial %%n) := n.to_nat
+| _ := none
+end
+
+/--  The function we use to compare two `expr`:
+* all non-monomials are compared alphabetically;
+* all non-monomials are smaller than all monomials;
+* bigger monomials have higher exponent.
+-/
+meta def compare_fn (eₗ eᵣ : expr) : bool :=
+match (monomial_weight eₗ, monomial_weight eᵣ) with
+| (none, none)     := eₗ.to_string ≤ eᵣ.to_string -- this solution forces an unique ordering
+| (_, none)        := false
+| (none, _)        := true
+| (some l, some r) := l ≤ r
+end
+
+/--  A version of `sort_summands_aux` that allows failure, if `allow_failure = tt`. -/
+meta def sort_summands_core (allow_failure : bool) (t : list expr) (hyp : option name) :
+  tactic unit :=
+if allow_failure then sort_summands_aux compare_fn t hyp <|> skip
+else sort_summands_aux compare_fn t hyp
+
+/-- If the target is an equality involving summands,
+then  `sort_summands` sorts the summands on either side of the equality. -/
+meta def sort_summands (l : parse pexpr_list_or_texpr?) (locat : parse location) : tactic unit :=
+do
+  l : list expr ← (l.get_or_else []).mmap to_expr,
+  match locat with
+  | loc.wildcard := do
+    sort_summands_core tt l none,
+    ctx ← local_context,
+    ctx.mmap (λ e, sort_summands_core tt l (expr.local_pp_name e)),
+    skip
+  | loc.ns names := do
+    names.mmap $ sort_summands_core ff l,
+    skip
+  end
+
```

Comment:
There might be some more applicable tags, check the list of tags by expanding "Filter by tag" in the right sidebar [here](https://leanprover-community.github.io/mathlib_docs/tactics.html). Also, since the docstring of `sort_summands` will become the tactic doc on that page, perhaps some of the useful info in the module doc should be copied there.
```suggestion

add_tactic_doc
{ name := "sort_summands",
  category := doc_category.tactic,
  decl_names := [`tactic.interactive.sort_summands],
  tags := ["arithmetic"] }

```

Reply:
Thanks Bryan!  I'll look for other applicable tags once I turn on my computer!

Reply:
I did not find another suitable tag, so I just left `arithmetic`, that you had suggested.